src/main/java/bio/terra/service/dataset/flight/upgrade/enableSecureMonitoring/EnableSecureMonitoringEnableFlagStep.java

```diff
@@ -0,0 +1,58 @@
+package bio.terra.service.dataset.flight.upgrade.enableSecureMonitoring;
+
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.service.dataset.DatasetDao;
+import bio.terra.service.dataset.flight.DatasetWorkingMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import java.util.UUID;
+
+public class EnableSecureMonitoringEnableFlagStep implements Step {
+  private final DatasetDao datasetDao;
+  private final AuthenticatedUserRequest userRequest;
+
+  public EnableSecureMonitoringEnableFlagStep(
+      DatasetDao datasetDao, AuthenticatedUserRequest userRequest) {
+    this.datasetDao = datasetDao;
+    this.userRequest = userRequest;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException {
+    FlightMap workingMap = context.getWorkingMap();
+    UUID datasetId = workingMap.get(DatasetWorkingMapKeys.DATASET_ID, UUID.class);
+    boolean patchSucceeded = datasetDao.setSecureMonitoring(datasetId, true, userRequest);
```

Comment:
Would it make sense to only do this patch if `DatasetWorkingMapKeys.SECURE_MONITORING_ENABLED` is false?

Reply:
That would definitely also be a valid approach. 
I think the use case y'all discussed at standup could make it worth it to keep the upgrade even if it's already true -- It will still go through and create the TPS policies.

Reply:
Oh I was referring to just this step. So if we know that a dataset already has secure monitoring enabled, we could skip this call to `datasetDao.setSecureMonitoring`. Not worth blocking the PR though!

Reply:
That is an excellent point. 
But, yeah, we're getting close to PR blocking territory. I've already done manual testing and I'd like for automated tests to run before merging today. And, while not optimal, it still should be okay without this change.